src/plugins/intel_cpu/src/nodes/subgraph.cpp

```diff
@@ -694,37 +694,40 @@ bool Snippet::SnippetJitExecutor::optimizeExecDomain(std::vector<VectorDims>& in
 }
 
 void Snippet::SnippetJitExecutor::generate(const jit_snippets_compile_args* jcp) {
-    ov::pass::Manager pre_dialect;
-    pre_dialect.register_pass<ConvertToSwishCPU>();
+    using Manager = snippets::pass::Manager;
+    std::vector<Manager::PositionedPass> backend_passes;
+
+#define SNIPPETS_ADD_POS_PASS(POS, PASS, ...) \
+            backend_passes.emplace_back(Manager::PassPosition(POS), std::make_shared<PASS>(__VA_ARGS__))
+
+    SNIPPETS_ADD_POS_PASS("", ConvertToSwishCPU);
     if (enforceBF16 && snippet_for_generation->has_domain_sensitive_ops()) {
         // enforce BF16 precisions to supported operations
         // MatMul has to be decomposed to Brgemm operations before enforcement
         // Note, MatMul decomposition will be ran later again for case if BF16 enforcement is not happened
-        CPU_REGISTER_PASS_X64(pre_dialect, ov::snippets::pass::MatMulToBrgemm);
-        CPU_REGISTER_PASS_X64(pre_dialect, pass::EnforcePrecision, element::f32, element::bf16);
+        SNIPPETS_ADD_POS_PASS("", ov::snippets::pass::MatMulToBrgemm);
+        SNIPPETS_ADD_POS_PASS("", pass::EnforcePrecision, element::f32, element::bf16);
     }
 
-    ov::pass::Manager post_dialect;
-    CPU_REGISTER_PASS_X64(post_dialect, ov::intel_cpu::pass::BrgemmToBrgemmCPU);
-    CPU_REGISTER_PASS_X64(post_dialect, ov::intel_cpu::pass::SetBrgemmCPUBlockingParams);
+    SNIPPETS_ADD_POS_PASS("PropagatePrecision", ov::intel_cpu::pass::BrgemmToBrgemmCPU);
+    SNIPPETS_ADD_POS_PASS("PropagatePrecision", ov::intel_cpu::pass::SetBrgemmCPUBlockingParams);
+
+    const auto& pipeline_end = Manager::PassPosition("", true);
```

Comment:
I would say usage of "' or bool flag is not very convenient way to describe the desired behavior. What do you you think about extension of PassPosition methods (with type enum or separate classes):
- PassPositionAfter(pass)
- PassPositionBefore(pass)
- PassPositionPipelineStart()
- PassPositionPipelineEnd()

Reply:
Ok, why not.
It's a matter of taste, the implementation through classes seems too bulky for me since you get 6 tiny-winy classes instead of one: interface + 4 that you mentioned + one intermediate for `PassPositionAfter` and `PassPositionBefore` to store pass_name, instance, and get_insert_pos implementation.
I therefore chose the enum path, please take a look.

Reply:
I am ok with the solution. Major result it looks much more readable now. Thanks!